app/src/main/java/org/jellyfin/mobile/media/car/LibraryBrowser.kt

```diff
@@ -245,143 +235,162 @@ class LibraryBrowser(
+            filters = listOf(ItemFilter.IS_PLAYED),
+            sortBy = listOf("DatePlayed"),
+            sortOrder = listOf(SortOrder.DESCENDING),
+            recursive = true,
+            imageTypeLimit = 1,
+            enableImageTypes = listOf(ImageType.PRIMARY),
+            enableTotalRecordCount = false,
             limit = 100
-        }
-        return apiClient.getItems(query)?.extractItems("${LibraryPage.RECENTS}|$libraryId")
+        )
+
+        return result.extractItems("${LibraryPage.RECENTS}|$libraryId")
     }
 
     private suspend fun getAlbums(
-        libraryId: String,
-        filterArtist: String? = null,
-        filterGenre: String? = null
+        libraryId: UUID,
+        filterArtist: UUID? = null,
+        filterGenre: UUID? = null
     ): List<MediaBrowserCompat.MediaItem>? {
-        val query = ItemQuery().apply {
-            userId = apiClient.currentUserId
-            parentId = libraryId
-            when {
-                filterArtist != null -> artistIds = arrayOf(filterArtist)
-                filterGenre != null -> genreIds = arrayOf(filterGenre)
-            }
-            includeItemTypes = arrayOf(BaseItemType.MusicAlbum.name)
-            sortBy = arrayOf(ItemSortBy.DatePlayed)
-            sortOrder = SortOrder.Descending
-            recursive = true
-            imageTypeLimit = 1
-            enableImageTypes = arrayOf(ImageType.Primary)
+        val result by itemsApi.getItems(
+            userId = appPreferences.currentUserUuid,
+            parentId = libraryId,
+            artistIds = filterArtist?.let(::listOf),
+            genreIds = filterGenre?.let(::listOf),
+            includeItemTypes = listOf("MusicAlbum"),
+            sortBy = listOf("DatePlayed"),
+            sortOrder = listOf(SortOrder.DESCENDING),
+            recursive = true,
+            imageTypeLimit = 1,
+            enableImageTypes = listOf(ImageType.PRIMARY),
             limit = 100
-        }
-        return apiClient.getItems(query)?.extractItems()?.browsable()
+        )
+
+        return result.extractItems()?.browsable()
     }
 
-    private suspend fun getArtists(libraryId: String): List<MediaBrowserCompat.MediaItem>? {
-        val query = ArtistsQuery().apply {
-            userId = apiClient.currentUserId
-            parentId = libraryId
-            sortBy = arrayOf(ItemSortBy.SortName)
-            sortOrder = SortOrder.Ascending
-            recursive = true
-            imageTypeLimit = 1
-            enableImageTypes = arrayOf(ImageType.Primary)
+    private suspend fun getArtists(libraryId: UUID): List<MediaBrowserCompat.MediaItem>? {
+        val result by artistsApi.getArtists(
+            userId = appPreferences.currentUserUuid,
+            parentId = libraryId,
+            // FIXME missing: sortBy = arrayOf(ItemSortBy.SortName)
+            // FIXME missing: sortOrder = SortOrder.Ascending
+            // FIXME missing: recursive = true
```

Comment:
These 3 properties don't exist in the current API, they probably didn't in the old one either so they can probably be removed without any issue.